Source/WebCore/rendering/RenderElement.h

```diff
@@ -294,6 +294,7 @@ class RenderElement : public RenderObject {
     bool createsNewFormattingContext() const;
 
     bool isSkippedContentRoot() const;
+    bool isSkippedContentRootForLayout() const;
```

Comment:
does this mean we have to call isSkippedContentRootForLayout() when inside layout as opposed to isSkippedContentRoot()? if so, how to tell whether you are _inside_ layout?

Reply:
Sorry, I got sidetracked with other tasks. Yes, isSkippedContentRootForLayout() is the safe variant to call when in layout. Usually the places to call this from are pretty clear, layoutBlock and layoutPositionedObject for example. Do you foresee call sites where it is not clear whether we are in layout or not? Regarding the grid changes, the expectation is that the track sizing algorithm is only called when laying out.